third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -3236,6 +3243,156 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         }
     }
 
+    fn call_to_borrow_global(
+        &mut self,
+        loc: &Loc,
+        e: &EA::Exp,
+        i: &EA::Exp,
+        mutable: bool,
+        expected_type: &Type,
+        context: &ErrorMessageContext,
+    ) -> ExpData {
+        fn convert_name_to_type(
+            loc: &move_ir_types::location::Loc,
+            exp_: EA::Exp_,
+        ) -> Option<EA::Type> {
+            if let EA::Exp_::Name(m, type_opt) = exp_ {
+                Some(EA::Type::new(
+                    *loc,
+                    EA::Type_::Apply(m, type_opt.unwrap_or(vec![])),
+                ))
+            } else {
+                None
+            }
+        }
+        let type_opt = convert_name_to_type(&e.loc, e.clone().value).map(|ty| vec![ty]);
+        let module_name = self.parent.parent.builtin_module();
+        let oper_name = if mutable {
+            "borrow_global_mut"
+        } else {
+            "borrow_global"
+        };
+        let symbol = self.parent.parent.builtin_qualified_symbol(oper_name);
+        let name_loc = &self.to_loc(&e.loc);
+        self.translate_call(
+            loc,
+            name_loc,
+            CallKind::Regular,
+            &Some(module_name.clone()),
+            symbol.symbol,
+            &type_opt,
+            &[i],
+            expected_type,
+            context,
+        )
+    }
+
+    fn call_to_vector_borrow(
+        &mut self,
+        loc: &Loc,
+        e: &EA::Exp,
+        i: &EA::Exp,
+        mutable: bool,
+        expected_type: &Type,
+        context: &ErrorMessageContext,
+    ) -> ExpData {
+        let mut symbol_opt = None;
+        let mut module_opt = None;
+        let target_str = if mutable {
+            "vector::borrow_mut"
+        } else {
+            "vector::borrow"
+        };
+        for (name, mid) in &self.parent.parent.module_table {
+            let name_str = name.name().display(self.symbol_pool()).to_string();
+            if name_str == "vector" {
```

Comment:
Do we also need to check the module address, or are we happy using the syntactic sugar for any module named vector?

Reply:
good point, let me check and fix if necessary.